src/bt_peer_connection.cpp

```diff
@@ -481,7 +481,13 @@ namespace {
 		if (support_extensions()) p.flags |= peer_info::supports_extensions;
 		if (is_outgoing()) p.flags |= peer_info::local_connection;
 #if TORRENT_USE_I2P
-		if (is_i2p(get_socket())) p.flags |= peer_info::i2p_socket;
+		if (is_i2p(get_socket()))
+		{
+			p.flags |= peer_info::i2p_socket;
+			auto const* pi = peer_info_struct();
+			sha256_hash const b32_hash = hasher256(pi->dest()).final();
```

Comment:
`b32_` prefix confuses here.

Reply:
my understanding is that this address is called "base32 address", even though the most interesting property of it is that it's the sha256 hash of the destination.

Reply:
> my understanding is that this address is called "base32 address"

Yes. But here is no Base32 address. This is a preform for it.